src/charm.py

```diff
@@ -83,8 +117,36 @@ def __init__(self, *args):
         self.framework.observe(self.ingress.on.ready, self._on_ingress_ready)
         self.framework.observe(self.ingress.on.revoked, self._on_ingress_revoked)
 
+        self.framework.observe(
+            self.loki_consumer.on.promtail_digest_error,
+            self._promtail_error,
+        )
+
+
+    def _on_install(self, event: InstallEvent) -> None:
+        if not self._container.can_connect():
+            event.defer()
+            self.unit.status = WaitingStatus("Waiting to connect to Login_UI container")
+            return
+
+        # initialize logging file
+
+        Path(self._log_dir).mkdir(parents=True, exist_ok=True)
+        Path(self._log_path).touch()
```

Comment:
This is going to be a bit awkward: Guillaume explained to me that creating the dir and logfile in the on_install handler is not a good idea. I don't remember the exact reasoning, but it's in the ballpark of when the k8s restarts a container, then when the container/pod gets redeployed it won't shoot the on_install event, and you'll be left without the dir and logfile.
I suggest putting this stuff in the _on_pebble_ready handler.

Reply:
so we have the rock creating the `/var/log` folder, i am not exactly clear if the log rotation creates the file if not there (which allegedly is the case) so this might be not needed, will move it anyway to the other hook

Reply:
gave it another spin locally and file gets created by the `logRotator`, will drop this